### PR TITLE
Add feedback component

### PR DESCRIPTION
**Pull request by Etesam913**
### Summary
Adds feedback component to `mephisto-worker-addons`. 

Properties for this component are defined in the `mephisto-worker-addons` readme.

There is a questions prop that allows for questions to be associated with a feedback textarea when reviewing.

Uses the handleSubmitMetadata() method from `mephisto-task` in the default handleSubmit() of the Feedback component to make submitting feedback very easy. **Very similar to how it is done in the Tips component** 

Adds feedback component to static_react_task_with_tips webapp

Created a reducer to add error checking for both the feedback & tips components.

Added some small changes like new properties for the Tips component. 

#### New video:
https://github.com/facebookresearch/Mephisto/pull/793#issuecomment-1184900265

#### Old Video:
https://user-images.githubusercontent.com/55665282/176503447-0d76ddd7-2d76-4a6d-8ac7-f4f1b9cbef1e.mov

In the video it should be clear that I write feedback for the two questions. The feedback for the first question is clearly not toxic, while the feedback for the second question clearly is.

Then I ran the review_feedback_for_task.py script where I was first able to filter by the question asked. Then I was able to filter by toxicity & then filter by unreviewed/reviewed feedback. 

I ran the script three times.

The first time I reviewed the feedback for the first question. The toxicity of the feedback was < 0.5, so filtering by toxicity would do nothing.

The second time I reviewed the feedback for the other question. I chose to filter by toxicity and the feedback message did not show up as it's toxicity > 0.5.

The third time I reviewed the same question as the second time, but this time I chose not to filter by toxicity. As expected, it showed the message.

### TODO:
- [x] Make detoxify get installed when pip install -e . is ran in root
- [x] Planning on publishing the changes to mephisto-task and mephisto-worker-addons after this pull request gets merged

### CODE REVIEW

**Comment by Rebecca-Qian** [comment]
This is a great concept, which we implemented for a Mephisto task with good results. I have a couple of design suggestions/questions:
- Why is it at the bottom right? If the goal is to make it look like a popup, you can add a background color and border, option to X out etc. Another option is to have it directly after task submission.
- You want to make it clear that this is entirely optional and won't affect worker pay. So with a title like "Task Feedback (optional)" or short description before the worker does it.
- I like the idea of filtering by toxicity, but it may also be helpful for researchers to know that some workers are particularly upset?
- Right now the UX looks very similar to tips - might be a good idea to differentiate between the two.

Really cool stuff!

**Comment by Etesam913** [comment]
I'm not exactly sure that the feedback component is supposed to look like a popup. It is unlikely that anyone would write feedback if it was hidden in some popup. The location of the items being in the bottom right is kind of weird so I just moved it to be right below the green and red buttons.

I definitely agree with making it clear that the feedback should be optional. It wasn't really clear in the way that the component looked.

Here are some of the changes that I made based off of your comments.

feedback-improvement

File: examples/static_react_task_with_tips/webapp/src/components/core_components.jsx
Comment (Etesam913) [inline examples/static_react_task_with_tips/webapp/src/components/core_components.jsx:97]: The new Feedback component, you can optionally \ass in questions or have no questions and just have general feedback.

File: mephisto/data_model/agent.py
Comment (Etesam913) [inline mephisto/data_model/agent.py:304]: New feedback object that gets added

File: mephisto/scripts/local_db/review_feedback_for_task.py
Comment (Etesam913) [inline mephisto/scripts/local_db/review_feedback_for_task.py:13]: Most of this script gets reworked with rich in #816

File: packages/mephisto-worker-addons/README.md
Comment (Etesam913) [inline packages/mephisto-worker-addons/README.md:89]: Added some new props for tips in this pr.

If length exceeds the max lengths an error message shows up that prevents the submission of the feedback

File: packages/mephisto-worker-addons/README.md
Comment (Etesam913) [inline packages/mephisto-worker-addons/README.md:118]: Feedback properties

File: packages/mephisto-worker-addons/src/Feedback/Question.js
Comment (Etesam913) [inline packages/mephisto-worker-addons/src/Feedback/Question.js:45]: These are a lot of props for a component, but using context or a state management lib are probably overkill for this package.

File: packages/mephisto-worker-addons/src/Feedback/index.js
Comment (Etesam913) [inline packages/mephisto-worker-addons/src/Feedback/index.js:51]: initialize with correct length

File: packages/mephisto-worker-addons/src/Feedback/index.js
Comment (Etesam913) [inline packages/mephisto-worker-addons/src/Feedback/index.js:33]: Handles different states that the feedback component can be in (error, loading, normal, etc...)

File: packages/mephisto-worker-addons/src/index.jsx
Comment (Etesam913) [inline packages/mephisto-worker-addons/src/index.jsx:3]: The createFeedback & createTip methods are way more involved/complex than they previously were, so it makes not sense for them to be exported for the user to run.

**Review by Etesam913 — COMMENTED** [review]
Added some explanatory comments on what I have done

**Comment by Etesam913** [comment]
Tests are failing because detoxify is not installed globally

**Comment by Etesam913** [comment]
Updated video:

https://user-images.githubusercontent.com/55665282/179086794-700c66e1-29b8-4717-92c3-b2c40a3becd3.mov

**Comment by JackUrb** [comment]
Can you merge #779 into this and then set it as a base branch for this PR? Should let you then add `detoxify` as a dependency in `pyproject.toml`.

**Comment by JackUrb** [comment]
`requirements.txt` no longer exists after that PR.

**Comment by Etesam913** [comment]
Okay, so I guess I just have to remember to switch the base branch back to main before merging this pr in.

File: mephisto/scripts/local_db/review_feedback_for_task.py
Comment (JackUrb) [inline mephisto/scripts/local_db/review_feedback_for_task.py:93]: As an aside, this style of task name selection is a pretty common utility, and would likely be a good candidate to abstract into the `mephisto.tools.scripts` collection. Can do after #816

File: mephisto/scripts/local_db/review_tips_for_task.py
Comment (JackUrb) [inline mephisto/scripts/local_db/review_tips_for_task.py:96]: Ideally for this and the `yes_no_responses` case, you should use set joining:
```
acceptable_responses = accept_response | reject_response
```

File: pyproject.toml
Comment (JackUrb) [inline pyproject.toml:54]: Just for organization, can you add this under a `# [tool.poetry.group.worker_addons.dependencies]` heading? I'll be toggling these in poetry 1.2.

**Review by JackUrb — COMMENTED** [review]
Only small comments on the backend, the overall implementation looks great! 

As an aside, I imagine it would be really cool to provide a navbar-style meta-component that can hold these types of add-ons behind button toggles in an attached overlay. (similar to how tips already works!)

I'll hold off on stamping to merge until tests are passing and the underlying PR is in.

File: mephisto/scripts/local_db/review_tips_for_task.py
Comment (JackUrb) [reply mephisto/scripts/local_db/review_tips_for_task.py:96]: Totally fair to wait for `rich`

File: mephisto/data_model/agent.py
Comment (Etesam913) [reply mephisto/data_model/agent.py:33]: That certainly is a way of doing it. Something that I saw in the codebase was putting imports inside an if statement
```python
elif "feedback" in data:
    from detoxify import Detoxify
    ...

Comment (JackUrb) [reply mephisto/data_model/agent.py:33]: I was considering this approach as well, though it makes the choice less deliberate. I tend to only use it when there are potential dependency loops.

**Review by JackUrb — APPROVED** [review]
I think this is ready to go. Thanks for the patience here in getting this right @Etesam913!

**Comment by Etesam913** [comment]
@pringshia I should probably make the non-headless version of this component be in a shadow-dom right?

**Pull request merged**
